### PR TITLE
feat: bump lua-resty-ldap version for ldap-auth

### PR DESCRIPTION
**Pull request by bzp2010**
### Description

A new `lua-resty-ldap` has been released that provides a completely different API and more new features, we need to upgrade it to exclude API incompatibilities and prepare for another advanced LDAP plugin.

This upgrade is completely non-destructive to the user, and theoretically any test cases do not need to be modified to prove compatibility.

Head of #9022

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/ldap-auth.lua [hunk 1]
```diff
@@ -100,6 +100,11 @@ local function extract_auth_header(authorization)
 end
 
 function _M.rewrite(conf, ctx)
+    if not ok then -- ensure rasn library loaded
```
Comment (spacewander) [inline apisix/plugins/ldap-auth.lua:103]: For code style, we return 501 in the other similar cases:
https://github.com/apache/apisix/blob/7e292eadd36e738f84647634b0c351bc268ea529/apisix/plugins/gzip.lua#L104-L108

File: ci/common.sh [hunk 2]
```diff
@@ -56,6 +56,19 @@ rerun_flaky_tests() {
     FLUSH_ETCD=1 prove --timer -I./test-nginx/lib -I./ $(echo "$tests" | xargs)
 }
 
+install_curl () {
+    wget https://curl.se/download/curl-7.88.0.tar.gz
```
Comment (spacewander) [inline ci/common.sh:60]: What about exacting the version out like the other install_xxx function?

Comment (bzp2010) [reply ci/common.sh]: Yes, it makes sense, fixed.

File: apisix/plugins/ldap-auth.lua (hunk 1 above)
Comment (bzp2010) [reply apisix/plugins/ldap-auth.lua:103]: changed

**Review by spacewander — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**